legislature/models.py

```diff
@@ -94,7 +64,44 @@ def votes_with_proposal(self):
         ).filter(representative=self)
 
 
-@receiver(post_save, sender=Representative)
-def create_memopolrepresentative_from_representative(instance, **kwargs):
-    memopol_representative = create_child_instance_from_parent(MemopolRepresentative, instance)
-    memopol_representative.save()
+def mempol_representative(sender, representative, data, **kwargs):
+    update = False
+    try:
+        memopol_representative = MemopolRepresentative.objects.get(
+            representative_ptr=representative)
+    except MemopolRepresentative.DoesNotExist:
+        memopol_representative = MemopolRepresentative(
+            representative_ptr=representative)
+
+        # Please forgive the horror your are about to witness, but this is
+        # really necessary. Django wants to update the parent model when we
+        # save a child model.
+        memopol_representative.__dict__.update(representative.__dict__)
+
+    try:
+        country = sorted(data.get('Constituencies', []),
+                         key=lambda c: c.get('end') if c is not None else 1
+                         )[-1]['country']
+    except IndexError:
+        pass
+    else:
+        if sender.cache.get('countries', None) is None:
+            sender.cache['countries'] = {c.name: c.pk for c in
+                                         Country.objects.all()}
+        country_id = sender.cache['countries'].get(country)
+
+        if memopol_representative.country_id != country_id:
+            memopol_representative.country_id = country_id
+            update = True
+
+    if sender.mep_cache['groups']:
+        main_mandate = sorted(sender.mep_cache['groups'],
+                              key=lambda m: m.end_date)[-1]
+
+        if memopol_representative.main_mandate_id != main_mandate.pk:
+            memopol_representative.main_mandate_id = main_mandate.pk
+            update = True
+
+    if update:
+        memopol_representative.save()
+representative_post_save.connect(mempol_representative)
```

Comment:
There should be 2 blanks lines around functions definition at the root level of a file, no?